src/backoff.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+// https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+
+use rand::{thread_rng, Rng};
+
+pub trait Backoff: Clone + Copy + Send + 'static {
+    fn next_delay_ms(&mut self) -> Option<u64>;
+}
+
+#[derive(Clone, Copy)]
```

Comment:
I don't think we need to mark it `Copy`. Explicit clones will show how the states are changed more clearly.